### ISSUE TITLE
mypy allows to override @final methods 

### ISSUE THREAD

**Issue by rkashapov** [python/mypy#14371]
**Bug Report**

mypy allows to override @final methods 

**To Reproduce**
The following code is reported as valid by mypy
```python
from typing import final


class Base:

    @final
    def foo(self):
        return 'foo'


class Child(Base):

    def foo(self):
        return 'bar'
```

**Expected Behavior**

```
$ mypy main.py
main.py:13: error: Cannot override final attribute "foo" (previously declared in base class "Base")  [misc]
```

**Actual Behavior**

```
$ mypy main.py
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: mypy 0.991 (compiled: yes)
- Mypy command-line flags: no flags
- Mypy configuration options from `mypy.ini` (and other config files): no config files
- Python version used: 3.11.0

**Comment by JelleZijlstra** [comment]
The function is unannotated, so mypy doesn't check it. See https://mypy.readthedocs.io/en/stable/common_issues.html#no-errors-reported-for-obviously-wrong-code.
